src/main/java/appeng/util/inv/AdaptorItemHandler.java

```diff
@@ -0,0 +1,247 @@
+package appeng.util.inv;
+
+
+import java.util.Iterator;
+
+import net.minecraft.item.ItemStack;
+import net.minecraftforge.items.IItemHandler;
+
+import appeng.api.config.FuzzyMode;
+import appeng.util.InventoryAdaptor;
+import appeng.util.Platform;
+
+
+public class AdaptorItemHandler extends InventoryAdaptor
+{
+	private final IItemHandler itemHandler;
+
+	public AdaptorItemHandler( IItemHandler itemHandler )
+	{
+		this.itemHandler = itemHandler;
+	}
+
+	@Override
+	public ItemStack removeItems( int amount, ItemStack filter, IInventoryDestination destination )
+	{
+		int slots = itemHandler.getSlots();
+		ItemStack rv = null;
+
+		for( int slot = 0; slot < slots && amount > 0; slot++ )
+		{
+			final ItemStack is = itemHandler.getStackInSlot( slot );
+			if( is == null || ( filter != null && !Platform.isSameItemPrecise( is, filter ) ) )
+			{
+				continue;
+			}
+
+			if( destination != null )
+			{
+				ItemStack extracted = itemHandler.extractItem( slot, amount, true );
+				if( extracted == null )
+				{
+					continue;
+				}
+
+				if( !destination.canInsert( extracted ) )
+				{
+					continue;
+				}
+			}
+
+			// Attempt extracting it
+			ItemStack extracted = itemHandler.extractItem( slot, amount, false );
+
+			if( extracted == null )
+			{
+				continue;
+			}
+
+			if( rv == null )
+			{
+				// Use the first stack as a template for the result
+				rv = extracted;
+				filter = extracted;
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Introduce a new variable instead of reusing the parameter "filter". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1226)